big_scape/distances/query.py

```diff
@@ -223,3 +184,9 @@ def callback(edges):
         bs_data.DB.commit()
 
         logging.info("Generated %d edges", num_edges)
+
+        if run["skip_propagation"]:
+            # in this case we only want one iteration, the Query -> Ref edges
+            break
+
+        bin.bin.cycle_records()
```

Comment:
better to add a cycle_records method to the missingpairgenerator and make this bin.cycle_records.

bin.bin should probably not be accessed directly like this